operator/src/main/java/org/bf2/operator/managers/StrimziManager.java

```diff
@@ -146,39 +169,65 @@ private void updateStatus() {
-        if (kafkaImagesEnvVar.isPresent() && kafkaImagesEnvVar.get().getValue() != null) {
-            String kafkaImages = kafkaImagesEnvVar.get().getValue();
-            String[] kafkaImagesList = kafkaImages.split("\n");
-            kafkaVersions = new ArrayList<>(kafkaImagesList.length);
-            kafkaIbpVersions = new ArrayList<>(kafkaImagesList.length);
-            for (String kafkaImage : kafkaImagesList) {
-                String kafkaVersion = kafkaImage.split("=")[0];
-                String kafkaIbpVersion = AbstractKafkaCluster.getKafkaIbpVersion(kafkaVersion);
-                kafkaVersions.add(kafkaVersion);
-                kafkaIbpVersions.add(kafkaIbpVersion);
-            }
-        }
+        StrimziVersionStatus strimziVersion = new StrimziVersionStatusBuilder()
+                .withVersion(deployment.getMetadata().getName())
+                .withKafkaVersions(kafkaVersions)
+                .withKafkaIbpVersions(kafkaIbpVersions)
+                .withReady(Readiness.isDeploymentReady(deployment))
+                .build();
 
         this.strimziVersions.put(deployment.getMetadata().getName(),
-                new StrimziVersionStatusBuilder()
-                        .withVersion(deployment.getMetadata().getName())
-                        .withKafkaVersions(kafkaVersions)
-                        .withKafkaIbpVersions(kafkaIbpVersions)
-                        .withReady(Readiness.isDeploymentReady(deployment))
-                        .build());
+                new ComponentVersions(strimziVersion, relatedImages));
+    }
+
+    List<String> readKafkaVersions(Deployment deployment) {
+        return readAnnotation(deployment, KAFKA_IMAGES_ANNOTATION)
+            .or(() -> readVersionEnvVar(deployment))
+            .map(this::parseKafkaVersions)
+            .orElseGet(Collections::emptyList);
+    }
+
+    Optional<String> readVersionEnvVar(Deployment deployment) {
+        return deployment.getSpec()
+                .getTemplate()
+                .getSpec()
+                .getContainers()
+                .stream()
+                .filter(container -> container.getName().startsWith(STRIMZI_CLUSTER_OPERATOR))
+                .flatMap(container -> container.getEnv().stream())
+                .filter(ev -> ev.getName().equals(KAFKA_IMAGES_ENVVAR))
+                .map(EnvVar::getValue)
+                .filter(Objects::nonNull)
+                .findFirst();
+    }
+
+    List<String> parseKafkaVersions(String versions) {
+        return Arrays.stream(versions.split("\n"))
+                .map(entry -> entry.split("="))
+                .map(entry -> entry[0])
+                .collect(Collectors.toCollection(ArrayList::new));
+    }
+
+    @SuppressWarnings("unchecked")
+    Map<String, String> readRelatedImagesAnnotation(Deployment deployment) {
+        return readAnnotation(deployment, RELATED_IMAGES_ANNOTATION)
+                .map(value -> Serialization.unmarshal(value, Map.class))
```

Comment:
The only doubt I have is about this unmarshall code, the map is defined as 
```
canary-init": "quay.io/mk-ci-cd/strimzi-canary/...
```
I know map works like
```
"canary-init" = "quay.io/mk-ci-cd/strimzi-canary/...
```
may be can we write unit test to verify this?

Reply:
`Serialization` is a wrapper around Jackson and it's parsed as JSON with a map format like `{ "key": "value" }`

This section added to the tests should cover it. Let me know what you think.
https://github.com/bf2fc6cc711aee1a0c2a/kas-fleetshard/pull/695/files#diff-c9ce9ec7624d9a9531bc0ad81e684c85f781ad02bc29b8277810c823ef3086b9R282-R284